tests/hardhat/Rewards.ts

```diff
@@ -39,11 +40,10 @@ let fakePriceOracle: FakeContract<PriceOracle>;
 let fakeAccessControlManager: FakeContract<AccessControlManager>;
 
 describe("Rewards: Tests", async function () {
-  const RewardsDistributor = await ethers.getContractFactory("RewardsDistributor");
   /**
    * Deploying required contracts along with the poolRegistry.
    */
-  before(async function () {
+  beforeEach(async function () {
```

Comment:
Might be quite slow. Maybe move it into a fixture?

Reply:
Done

Reply:
Done